hatchet/graphframe.py

```diff
@@ -112,18 +114,44 @@ def from_hpctoolkit(dirname):
     @staticmethod
     def from_hpctoolkit_latest(
         dirname: str,
+        directory_mapping: Dict[str, str] = None,
+        parallel_profiles_mode: bool = False,
         max_depth: int = None,
         min_percentage_of_application_time: int = None,
-        min_percentage_of_parent_time: int = None,
+        exclude_mpi_function_details: bool = False,
+        exclude_openmp_function_details: bool = False,
+        exclude_cuda_function_details: bool = False,
+        exclude_system_libraries_source_code: bool = False,
+        exclude_function_call_lines: bool = False,
+        exclude_no_source_code_instructions: bool = False,
+        exclude_instructions: bool = False,
+        exclude_non_function_nodes: bool = False,
+        label_function_nodes: bool = True,
+        metric_names: List[str] = None,
+        metric_scopes: List[str] = None,
+        summary_metrics: List[str] = None,
+        profile_ranks: List[int] = None,
     ):
         """
         Read an HPCToolkit database directory into a new GraphFrame
 
         Arguments:
             dirname (str): directory of an HPCToolkit performance database
-            max_depth (int): maximum depth that nodes in the CCT can have to be imported in Hatchet
-            min_percentage_of_application_time (int): minimum percentage of application time that nodes in the CCT must have to be imported in Hatchet
-            min_percentage_of_parent_time (int): minimum percentage of parent time that nodes in the CCT must have to be imported in Hatchet
+            directory_mapping (dict): Python dictionary that maps file system location to a name
+            parallel_profiles_mode (bool): fleg whether the reader should extract parallel profiles from the database (true) or the summary profile (false)
+            max_depth (int): maximum depth that nodes in the tree should have
+            min_percentage_of_application_time (int): minimum percentage of the application time that nodes in the tree should have
+            exclude_mpi_function_details (bool): fleg whether to exclude subtrees of MPI function nodes
```

Comment:
Even with `exclude_mpi_function_details` I see MPI nodes:
```
gf = ht.GraphFrame.from_hpctoolkit_latest(
        dirname=f"hatchet/tests/data/hpctoolkit-gamess",
        max_depth=10,
        exclude_mpi_function_details=True,
 )
```
Ex:
```
  │     ├─ 0.016 DDI_Finalize
  │     │  └─ 0.016 line [gamess.00.x]:0
  │     │     └─ 0.016 MPI_Finalize
```
and
```
...
└─ 803.289 line [gamess.00.x]:0
  ├─ 0.081 PMPI_Bcast
  └─ 803.208 DDI_Server
     └─ 803.208 loop [gamess.00.x]:0
```
It does work partially as I see it removing some nodes elsewhere in the tree:
```
{'type': 'function', 'name': 'PMPI_Ssend'}
{'type': 'function', 'name': 'MPI_Finalize'}
{'type': 'function', 'name': 'PMPI_Bcast'}
```

Also, the functionality changes whether `label_function_nodes` is `True` or `False` I think because the check [here](https://github.com/LLNL/hatchet/pull/141/files#diff-7e926d74e7ab2bf4c7dc20853f0c73bf690a4595077d6a525db9614f43425407R297-R298) will not be the same.

Reply:
@michaelmckinsey1 I tested this database. Flag `exclude_mpi_function_details` removes ONLY the subtree of an MPI function, meaning the MPI function itself will stay in the tree, but its subtree (which can be huge) will be removed. As I can see from the examples you provided, that is the case - MPI function subtrees are absent, but functions themselves are in the tree.